### DISCUSSION TITLE
I can't Commit to the main

### DISCUSSION THREAD

**Discussion by Aldebaran38** [community/community#30185] [New to GitHub]
When i click to Commit to main, Github Desktop give me this error:
![image](https://user-images.githubusercontent.com/77065456/185740397-9f5d16ad-895f-4e62-8f87-82617f24635e.png)

The list continues with the things i wantted to commit.
I don't know how to solve this or use the commands it gave me. Does anyone know how to solve this?

**Comment by airtower-luna** [comment]
The error message says "Changes not staged for commit". Before making a commit, you have to tell Git which changes you want to include in the commit. That's called *staging* them, and as the error message tells you [`git add <file>`](https://git-scm.com/docs/git-add) is the command to do it. Without any staged changes you can't commit because an empty commit doesn't make much sense.

So stage the changes you want, and then you should be able to commit. :slightly_smiling_face: 

See also: [Git Basics - Recording Changes to the Repository](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository)

**Reply by Aldebaran38** [reply]
How can i staging the the file that i want to commit? I select all of them.
![image](https://user-images.githubusercontent.com/77065456/185742268-2706038d-ca65-4251-b556-01f5819f4a88.png)
And also where should i enter these commands, and which ones?

**Reply by airtower-luna** [reply]
To stage using the command line you need a shell (exactly how depends on your OS), `cd` to the directory, and then run the commands (e.g. `git add` to stage files).

I assume GitHub Desktop also has some menu items to do it, but I'm not familiar with it. If you show a larger screenshot with menu options I might have some idea, or someone who uses GitHub Desktop might your post.

**Reply by Aldebaran38** [reply]
![image](https://user-images.githubusercontent.com/77065456/185742954-f309ccf5-d1d0-4aae-ab16-2958a82b5b36.png)
![image](https://user-images.githubusercontent.com/77065456/185743021-973b6547-a2ac-4a26-99ea-649382676c0c.png)
![image](https://user-images.githubusercontent.com/77065456/185743045-fedc9640-0c85-45cf-b0ff-0ca4f2327ade.png)

There is so much tabs, but i guess these are the important ones that can help.
@airtower-luna

**Reply by airtower-luna** [reply]
The config tabs aren't actually important, try right-clicking on a file in the file list.

The last picture implies that there's some way to get a command line, but again I'd expect that to be part of a context menu (the kind you right-click to get).

**Reply by Aldebaran38** [reply]
![image](https://user-images.githubusercontent.com/77065456/185743259-bf05ec70-0e06-4f1c-9076-c6e1b5272743.png)

Btw, when i try to preview a file it wont show up, even the .gitignore itself.
Other files, it says "No Changes found" and in .gitignore it says this file is empty

**Reply by airtower-luna** [reply]
Hm, that's really odd UI. :thinking: 

According to the docs you actually use those checkmarks to select what should be included in a commit: https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/making-changes-in-a-branch/committing-and-reviewing-changes-to-your-project

**Reply by Aldebaran38** [reply]
I try to do what the docs show, but i accidentally revert the things i do. How can i turn them back? @airtower-luna

**Reply by airtower-luna** [reply]
As I said above I'm not familiar with GitHub Desktop, I've been trying to guess, and evidently it's not working. Sorry about that.

I know *Git*, so I might be able to help if you try the command line instead. Otherwise I hope someone who knows GitHub Desktop see this post and can help.

**Reply by Aldebaran38** [reply]
I want to try command lines but still can't find where are they, and how do i use them.

**Reply by airtower-luna** [reply]
You seem to be on Windows, so something called "Git Bash" should have been installed along with Git. Search for that. The [Git documentation says](https://git-scm.com/book/en/v2/Getting-Started-The-Command-Line) Command Prompt or PowerShell should work, too, but Bash is probably best if you have it.

**Reply by Aldebaran38** [reply]
I don't have Git Bash on my PC. Only Git program i had was Git Desktop. But i install Git Bash from the site you showed me.
It open a tab that looks like CMD, i try "git add" and other commands but nothing happened.

**Reply by airtower-luna** [reply]
That looks like a good start. The first thing you need to do is switch to the folder where you have your repository (using `cd`). Then run `git status` to, well, check the status. It should show you on which branch the repository is, and a lists of changed/new files.

**Reply by Aldebaran38** [reply]
I try to check git status, but it gave me this error:
fatal: not a git repository (or any of the parent directories): .git
I use "cd" command before this.

**Reply by airtower-luna** [reply]
Just `cd` takes you to your user directory. You need to give it the path to the directory where you actually have the repository. I don't know where your repository is, so I can't say what exactly it'd look like, but maybe something similar to `cd programming/my-project`.

**Reply by Aldebaran38** [reply]
I write the location where my project is, but this time it gave me this error:
"  bash: syntax error near unexpected token `('  "

**Reply by airtower-luna** [reply]
Parentheses have special meanings in Bash. If there is an actual `(` in a directory name you need to do one of two things:
* *Escape* it (that means to tell Bash to treat it as a regular character) by adding backslash in front of it, like `\(`. Same for other characters with special meaning, like spaces.
* *Quote* the path, that means literally enclose it in quotes, like so: `"your path/with(special/characters"`

**Reply by Aldebaran38** [reply]
@airtower-luna There no need for Command line any more, i fixed.
I actually.. search the error wrong. I just search the error message, like i usually do. But since i search why GitHub Desktop never fetching, and find the answer. This is the post i fond:
https://stackoverflow.com/questions/68051949/github-desktop-suddenly-not-fetching-origin-cant-push-changes
Basicly just remove and re-add the repository...

**Reply by airtower-luna** [reply]
That's odd, but I'm glad you figured it out! :sweat_smile:

**Reply by queenofcorgis** [reply]
Thanks for sharing this was resolved @Aldebaran38 and thanks for lending a hand @airtower-luna ❤️ :octocat:
